tempesta_fw/connection.h

```diff
@@ -164,10 +164,9 @@ static inline bool
 tfw_connection_put(TfwConnection *conn)
 {
 	int rc;
-
-	if (unlikely(!conn))
-		return false;
-
+	rc = atomic_read(&conn->refcnt);
+	if (unlikely(!rc || rc == TFW_CONN_DEATHCNT))
+		return true;
```

Comment:
What if exactly at this point somebody `put()`s the connection? So first `rc` check fail and we're decrementing zero refcounter.